File: frontend/src/types/transaction.ts
// Transaction types
export type TxStatus = 'pending' | 'validated' | 'failed'

export interface Transaction {
  txId: string
  hash: string
  from: string
  to: string
  amount: number
  coin: string
  fee: number
  status: TxStatus
  timestamp: string
  validatedAt?: string
  blockNumber?: number
  note?: string
}

export interface ValidationLog {
  logId: string
  type: 'address' | 'transaction'
  input: string
  result: 'valid' | 'invalid' | 'suspicious'
  riskScore: number
  coin?: string
  checkedAt: string
  details?: Record<string, unknown>
}

export interface SavedAddress {
  addressId: string
  label: string
  address: string
  coin: string
  note?: string
  isValid: boolean
  riskScore: number
  validatedAt: string
}

export interface ValidationResult {
  address: string
  isValid: boolean
  riskScore: number
  flags: string[]
  network: string
  type: string
}

File: backend/app/models/transaction.py
from pydantic import BaseModel
from typing import Optional
from enum import Enum


class TxStatus(str, Enum):
    pending = "pending"
    validated = "validated"
    failed = "failed"


class TransactionCreate(BaseModel):
    hash: str
    from_address: str
    to_address: str
    amount: float
    coin: str
    fee: float = 0.0
    note: Optional[str] = None


class Transaction(TransactionCreate):
    txId: str
    status: TxStatus = TxStatus.pending
    timestamp: str
    validatedAt: Optional[str] = None
    blockNumber: Optional[int] = None


class UpdateStatusRequest(BaseModel):
    status: TxStatus


class ValidateAddressRequest(BaseModel):
    address: str
    coin: str


class BulkValidateRequest(BaseModel):
    addresses: list[ValidateAddressRequest]


class ValidateTransactionRequest(BaseModel):
    hash: str
    network: str


class SavedAddressCreate(BaseModel):
    label: str
    address: str
    coin: str
    note: Optional[str] = None
